core/src/main/java/io/questdb/cutlass/text/SerialCsvFileImporter.java

```diff
@@ -0,0 +1,147 @@
+        this.ff = configuration.getFilesFacade();
+        this.textLoader = new TextLoader(engine);
+        this.securityContext = AllowAllCairoSecurityContext.INSTANCE;
+    }
+
+    @Override
+    public void close() {
+        path = Misc.free(path);
+        textLoader = Misc.free(textLoader);
+    }
+
+    public void of(String tableName, String fileName, int partitionBy, CharSequence timestampColumn, boolean isHeaderFlag, int atomicity, ExecutionCircuitBreaker circuitBreaker) {
+        this.tableName = tableName;
+        this.fileName = fileName;
+        this.partitionBy = partitionBy;
+        this.timestampColumn = timestampColumn;
+        this.isHeaderFlag = isHeaderFlag;
+        this.atomicity = atomicity;
+        this.circuitBreaker = circuitBreaker;
+    }
+
+    public void process() throws TextImportException {
+        long buf = Unsafe.malloc(sqlCopyBufferSize, MemoryTag.NATIVE_PARALLEL_IMPORT);
+        setupTextLoaderFromModel();
+        path.of(sqlCopyInputRoot).concat(fileName).$();
+        long fd = ff.openRO(path);
+        try {
+            statusReporter.report(TextImportTask.PHASE_ANALYZE_FILE_STRUCTURE, TextImportTask.STATUS_STARTED, null);
+            if (fd == -1) {
+                throw TextImportException.instance(TextImportTask.PHASE_ANALYZE_FILE_STRUCTURE, "could not open file [errno=").put(Os.errno()).put(", path=").put(path).put(']');
+            }
+
+            long fileLen = ff.length(fd);
+            long n = ff.read(fd, buf, sqlCopyBufferSize, 0);
+            if (n > 0) {
+                textLoader.setForceHeaders(isHeaderFlag);
+                textLoader.setSkipRowsWithExtraValues(false);
+                textLoader.parse(buf, buf + n, securityContext);
+                textLoader.setState(TextLoader.LOAD_DATA);
+                int read;
+                statusReporter.report(TextImportTask.PHASE_ANALYZE_FILE_STRUCTURE, TextImportTask.STATUS_FINISHED, null);
+                statusReporter.report(TextImportTask.PHASE_PARTITION_IMPORT, TextImportTask.STATUS_STARTED, null);
+                while (n < fileLen) {
+                    if (circuitBreaker.checkIfTripped()) {
+                        TextImportException ex = TextImportException.instance(TextImportTask.PHASE_PARTITION_IMPORT, "import was cancelled");
+                        ex.setCancelled(true);
+                        throw ex;
+                    }
+                    read = (int) ff.read(fd, buf, sqlCopyBufferSize, n);
+                    if (read < 1) {
+                        throw TextImportException.instance(TextImportTask.PHASE_PARTITION_IMPORT, "could not read file [errno=").put(ff.errno()).put(']');
+                    }
+                    textLoader.parse(buf, buf + read, securityContext);
+                    n += read;
+                }
+                textLoader.wrapUp();
+                statusReporter.report(TextImportTask.PHASE_PARTITION_IMPORT, TextImportTask.STATUS_FINISHED, null);
```

Comment:
Would be great to have some `import complete ...` logging here similar to what we have in the parallel importer.

Reply:
yes, with this I'm waiting for your PR to be merged.